kinto/views/permissions.py

```diff
@@ -0,0 +1,64 @@
+from cornice import Service
+from pyramid.security import NO_PERMISSION_REQUIRED, Authenticated
+
+from kinto.authorization import PERMISSIONS_INHERITANCE_TREE
+from kinto.core import utils as core_utils
+
+
+permissions = Service(name='permissions',
+                      description='List of user permissions',
+                      path='/permissions')
+
+
+@permissions.get(permission=NO_PERMISSION_REQUIRED)
+def permissions_get(request):
+    # Invert the permissions inheritance tree.
+    perms_descending_tree = {}
+    for obtained, obtained_from in PERMISSIONS_INHERITANCE_TREE.items():
+        on_resource, obtained_perm = obtained.split(':', 1)
+        for from_resource, perms in obtained_from.items():
+            for perm in perms:
+                perms_descending_tree.setdefault(from_resource, {})\
+                                     .setdefault(perm, {})\
+                                     .setdefault(on_resource, set())\
+                                     .add(obtained_perm)
+
+    # Obtain current principals.
+    principals = request.effective_principals
+    if Authenticated in principals:
+        # XXX: since view does not require permission, had to do this:
+        userid = request.prefixed_userid
+        principals.append(userid)
+
+    # Query every possible permission from backend.
+    # XXX: this is a workaround because there is no "full-list" method.
+    possible_perms = set([k.split(':', 1)[1]
+                          for k in PERMISSIONS_INHERITANCE_TREE.keys()])
+    backend = request.registry.permission
+    perms_by_object_uri = {}
+    for perm in possible_perms:
+        object_uris = backend.principals_accessible_objects(principals, perm)
+        for object_uri in object_uris:
+            perms_by_object_uri.setdefault(object_uri, []).append(perm)
+
+    entries = []
+    for object_uri, perms in perms_by_object_uri.items():
+        # Obtain associated resource from object URI
+        resource_name, matchdict = core_utils.view_lookup(request, object_uri)
+        # XXX: do we want bucket_id, record_id or just id ?
+        matchdict['%s_id' % resource_name] = matchdict.pop('id')
```

Comment:
@n1k0 : 
this (consistent with notifications payload format):
```js
{
    "data": [
        {
            "bucket_id": "2f9b1aaa-552d-48e8-1b78-371dd08688b3", 
            "collection_id": "test", 
            "permissions": [
                "write", 
                "read", 
                "record:create"
            ], 
            "resource_name": "collection", 
            "uri": "/buckets/2f9b1aaa-552d-48e8-1b78-371dd08688b3/collections/test"
        }, 
        {
            "bucket_id": "2f9b1aaa-552d-48e8-1b78-371dd08688b3", 
            "permissions": [
                "write", 
                "read", 
                "collection:create", 
                "group:create"
            ], 
            "resource_name": "bucket", 
            "uri": "/buckets/2f9b1aaa-552d-48e8-1b78-371dd08688b3"
        }
    ]
}

```

or that (always an `id` attribute, whatever the resource type):
```js
{
    "data": [
        {
            "bucket_id": "2f9b1aaa-552d-48e8-1b78-371dd08688b3", 
            "id": "test", 
            "permissions": [
                "write", 
                "read", 
                "record:create"
            ], 
            "resource_name": "collection", 
            "uri": "/buckets/2f9b1aaa-552d-48e8-1b78-371dd08688b3/collections/test"
        }, 
        {
            "id": "2f9b1aaa-552d-48e8-1b78-371dd08688b3", 
            "permissions": [
                "write", 
                "read", 
                "collection:create", 
                "group:create"
            ], 
            "resource_name": "bucket", 
            "uri": "/buckets/2f9b1aaa-552d-48e8-1b78-371dd08688b3"
        }
    ]
}


```